jobs/telemetry_batch_view.py

```diff
@@ -22,6 +22,42 @@ def retrieve_jar():
     if jar_url is None:
         exit(1)
 
+
+    print("Retrieving JAR: {}".format(jar_url))
+
+    # Check to see if this is an alias for a full jar path
+    # If it's an alias, it should be accompanied by a .txt
+    # file whose contents point to the aliased location.
+    #
+    # The associated .txt files have two lines [0]:
+    # 1. The query string to get to the aliased jar
+    # 2. The associated build URL for that jar
+    #
+    # Historical version only had the query string [1],
+    # so we need to handle that case separately.
+    #
+    # [0] https://github.com/mozilla/telemetry-batch-view/blob/master/.circleci/deploy.sh#L37
+    # [1] https://github.com/mozilla/telemetry-batch-view/blob/14741db20dd3873b94944b8238dfc48a003c744d/deploy.sh#L50
+
+    txt_url = jar_url.replace(".jar", ".txt")
+    response = requests.get(txt_url)
+
+    if response.status_code != 404:
+        stripped = response.content.strip()
+        if len(stripped.split("\n")) == 1:
+            # Handle historical version
+            uri_query = stripped
+            build_url = "Build URL not available"
+        else:
+            # Output both query string and build URL
+            uri_query, build_url = stripped.split("\n")
```

Comment:
This would be slightly cleaner and safer using `.partition()`:

```
uri_query, _, build_url = response.content.strip().partition('\n')
if build_url is None:
  # Handle historical version
  build_url = "Build URL not available"
```

Reply:
Good recommendation!